mono/metadata/threads.c

```diff
@@ -1534,9 +1557,12 @@ ves_icall_System_Threading_Thread_Sleep_internal(gint32 ms)
 		mono_thread_clr_state (thread, ThreadState_WaitSleepJoin);
 
 		if (alerted) {
-			MonoException* exc = mono_thread_execute_interruption ();
-			if (exc) {
-				mono_set_pending_exception (exc);
+			MonoExceptionHandle exc = mono_thread_execute_interruption_handle (error);
+			mono_error_assert_ok (error);
+			if (exc && !MONO_HANDLE_IS_NULL (exc)) {
```

Comment:
a `MonoExceptionHandle` shouldn't be null.

Reply:
Low memory?

Reply:
We don't care about low memory. It's never been a design choice we have made and so it will take much more than this check to make it work.